src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -182,6 +184,21 @@ public ServerStore(RavenConfiguration configuration, RavenServer server)
 
             if (Configuration.Indexing.MaxNumberOfConcurrentlyRunningIndexes != null)
                 ServerWideConcurrentlyRunningIndexesLock = new FifoSemaphore(Configuration.Indexing.MaxNumberOfConcurrentlyRunningIndexes.Value);
+
+            CatastrophicFailureNotification = new CatastrophicFailureNotification((envId, path, exception, stacktrace) =>
+            {
+                try
+                {
+                    if (Logger.IsOperationsEnabled)
+                        Logger.OperationsAsync($"Catastrophic failure in server storage located at '{path}', StackTrace:'{stacktrace}'", exception).Wait(TimeSpan.FromSeconds(1));
```

Comment:
I'm waiting up to 1 second for this to be logged. Is that ok?

Reply:
Yes, that is fine.